docs/USING_PRO.md

```diff
@@ -60,10 +60,12 @@ Marked provides methods for directly overriding the `renderer` and `tokenizer` f
 
 <h2 id="renderer">The Renderer : <code>renderer</code></h2>
 
-The renderer defines the HTML output of a given token. If you supply a `renderer` object to the Marked options, it will be merged with the built-in renderer and any functions inside will override the default handling of that token type.
+The renderer defines the HTML output of a given token. If you supply a `renderer` in the options object passed to `marked.use()`, any functions in the object will override the default handling of that token type.
 
 Calling `marked.use()` to override the same function multiple times will give priority to the version that was assigned *last*. Overriding functions can return `false` to fall back to the previous override in the sequence, or resume default behavior if all overrides return `false`. Returning any other value (including nothing) will prevent fallback behavior.
 
+Note that a renderer created with `new Renderer()` will not be created with the default token handlers.
```

Comment:
What do you mean by this? Are you saying that using `marked.use({ renderer: new Renderer() });` will not use the default renderers, but it will.

Reply:
I think you are confused between the difference between `marked.use()` and `marked.setOptions()` or passing options directly to marked like `marked.parse('...', options)`.

`marked.use()` is for creating extensions including extending the renderer and tokenizer by providing an object.  
If you pass an object like:

```js
marked.use({ renderer: {
  em(text) {
    return `<em class="my-em-class">${text}</em>`
  }
}})
```

the `em` renderer will be overwritten but all other renderers will use the default.

`marked.setOptions()` and `marked.parse('...', options)` are for setting options.  
If you pass an object like:

```js
marked.parse('...', { renderer: {
  em(text) {
    return `<em class="my-em-class">${text}</em>`
  }
}})
```

the whole renderer will be overwritten and it will fail to render tokens other than `em`.

To override the renderer in the options you will have to extend `Renderer` like:

```js
import { marked, Renderer } from 'marked';

class MyRenderer extends Renderer {
  em(text) {
    return `<em class="my-em-class">${text}</em>`
  }
};

marked.parse('...', { renderer: new MyRenderer() });
```

Both are used for different things and have different benefits.

Reply:
Yes sorry, you're correct @UziTech. This line should go.

Reply:
```js
class MyRenderer extends Renderer {
  em(text) {
    return `<em class="my-em-class">${text}</em>`
  }
};
```

Maybe this part is obvious to others, but I was scratching my head trying to extend the renderer.

Perhaps it's also worth including in the docs, if it isn't there already.

Reply:
Another way that is commonly used to extend the renderer is to just reassign the renderer functions like:

```js
const renderer = new Renderer();
renderer.em = (text) => {
  return `<em class="my-em-class">${text}</em>` 
};

marked.parse('...', { renderer });
```

I think we want people to use `marked.use()` and move people away from using `marked.setOptions()` and `marked.parse('...', options)`.

Once we have a way to create an instance of marked (#907) I think those methods will be deprecated.

Reply:
Makes sense. I was trying to avoid `marked.use` for the same reason listed in that issue.